daal4py/bench.py

```diff
@@ -212,23 +233,71 @@ def time_mean_min(func, *args, inner_loops=1, outer_loops=1, **kwargs):
         'Must time the function at least once'
 
     times = np.zeros(outer_loops, dtype='f8')
+    total_time = 0.
 
-    for i in range(outer_loops):
-
-        t0 = timeit.default_timer()
+    # Warm-up iterations to determine optimal inner_loops
+    warmup = (goal_outer_loops > 0)
+    warmup_time = 0.
+    last_warmup = 0.
+    if warmup:
         for _ in range(inner_loops):
+            t0 = timeit.default_timer()
             val = func(*args, **kwargs)
-        t1 = timeit.default_timer()
-
-        times[i] = t1 - t0
+            t1 = timeit.default_timer()
+
+            warmup_time += t1 - t0
```

Comment:
I would reorder these, to save result of `t1-t0`, like so:

```
last_warmup = t1 - t0
warmup_time += last_warmup
```

Reply:
Done, thanks!